src/agent/src/main.rs

```diff
@@ -522,6 +512,31 @@ async fn init_attestation_components(logger: &Logger, config: &AgentConfig) -> R
     Ok(())
 }
 
+// Start-up attestation-agent, CDH and api-server-rest if they are packaged in the rootfs
+// and the corresponding procs are enabled in the agent configuration. the process will be
+// launched in the background and the function will return immediately.
+// If the CDH is started, a CDH client will be instantiated and returned.
+async fn init_attestation_components(logger: &Logger, config: &AgentConfig) -> Result<()> {
+    launch_guest_component_procs(logger, config).await?;
+
+    fs::write(OCICRYPT_CONFIG_PATH, OCICRYPT_CONFIG.as_bytes())?;
+    env::set_var("OCICRYPT_KEYPROVIDER_CONFIG", OCICRYPT_CONFIG_PATH);
+
+    // If a CDH socket exists, initialize the CDH client
+    match tokio::fs::metadata(CDH_SOCKET).await {
+        Ok(md) => {
+            if md.file_type().is_socket() {
+                cdh::init_cdh_client(CDH_SOCKET_URI).await?;
+            } else {
+                debug!(logger, "File {} is not a socket", CDH_SOCKET);
+            }
+        }
+        Err(err) => debug!(logger, "Failed to probe file {}: {:?}", CDH_SOCKET, err),
```

Comment:
Should this be using error! macro?

Reply:
hmm, I wouldn't say so. Is it a critical error if the socket doesn't exist? In this case we should return an error, no?

Reply:
In case of missing socket the CDH functionality will not be available and will impact sealed secrets and secure mount for now. Afaik, it won't affect other functionalities like signed/encrypted images. So may be logging the message (without requiring to enable debug) should be good. Probably warning message is better suited here?